core/templates/dev/head/domain/objects/NumberWithUnitsObjectFactory.js

```diff
@@ -119,23 +127,32 @@ oppia.factory('NumberWithUnitsObjectFactory', [
             value = rawInput.substr(0, ind).trim();
             units = rawInput.substr(ind).trim();
           }
+
+          if (units.indexOf('$') !== -1 || units.indexOf('Rs') !== -1 ||
```

Comment:
I think the whole idea of introducing currency units as a constant list was to eliminate this iffy condition which is too specific to a few particular units.

Reply:
done